packages/@ourworldindata/grapher/src/modal/DownloadModal.tsx

```diff
@@ -337,92 +375,487 @@ export class DownloadModal extends React.Component<DownloadModalProps> {
+            .filter((s) => s !== undefined && s.dataPublishedBy !== undefined)
+            .map(
+                (s): OwidOrigin => ({
+                    producer: s.dataPublishedBy,
+                    urlMain: s.link,
+                })
+            ) ?? []
+
+    const originsUniq = uniqBy(
+        [...origins, ...otherSources],
+        (o) => o.urlMain ?? o.datePublished
+    )
+
+    const attributions = getOriginAttributionFragments(originsUniq)
+
+    const sourceLinks = zip(attributions, originsUniq).map(
+        ([attribution, origin]) => {
+            const link = origin?.urlMain
+
+            if (link)
+                return (
+                    <li key={link}>
+                        <a href={link}>{attribution}</a>
+                    </li>
+                )
+            else return <li key={attribution}>{attribution}</li>
+        }
+    )
+
+    // Find the highest processing level of all columns
+    const owidProcessingLevel = table?.columnsAsArray
+        .map((col) => (col.def as OwidColumnDef).owidProcessingLevel)
+        .reduce((prev, curr) => {
+            if (prev === "major" || curr === "major") return "major" as const
+            if (prev === "minor" || curr === "minor") return "minor" as const
+            return undefined
+        }, undefined)
+
+    const sourceIsOwid =
+        attributions.length === 1 &&
+        attributions[0].toLowerCase() === "our world in data"
+    const processingLevelPhrase = !sourceIsOwid
+        ? getPhraseForProcessingLevel(owidProcessingLevel)
+        : undefined
+    const fullProcessingPhrase = processingLevelPhrase ? (
+        <>
+            {" "}
+            – <i>{processingLevelPhrase} by Our World In Data</i>
+        </>
+    ) : undefined
+
+    return (
+        <div className="download-modal__data-section download-modal__sources">
+            <h3 className="grapher_h3-semibold">Source and citation</h3>
+            {sourceLinks.length > 0 && (
+                <div className="download-modal__data-sources">
+                    <strong>Data sources:</strong>{" "}
+                    <ul className="download-modal__data-sources-list">
+                        {sourceLinks}
+                    </ul>
+                    {fullProcessingPhrase}
+                </div>
+            )}
+            <div>
+                <strong>Citation guidance:</strong> Please credit all sources
+                listed above. Data provided by third-party sources through Our
+                World in Data remains subject to the original{" "}
+                {sourceLinks.length === 1 ? "provider's" : "providers'"} license
+                terms.
+            </div>
+        </div>
+    )
+}
+
+const ApiAndCodeExamplesSection = (props: {
+    downloadCtxBase: DataDownloadContextBase
+    firstYColDef?: OwidColumnDef
+}) => {
+    const [onlyVisible, setOnlyVisible] = useState(false)
+    const [shortColNames, setShortColNames] = useState(false)
```

Comment:
```suggestion
    const [shortColNames, setShortColNames] = useState(true)
```

I think it's nicer by default to give code examples and download URLs with short names